### ISSUE TITLE
Storage account registration and Azure File Share Backup

### ISSUE THREAD

**Issue by pier62350** [hashicorp/terraform-provider-azurerm#14056]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform version: v0.15.5
AzureRM Provider: v2.84.0

### Affected Resource(s)

azurerm_backup_protected_file_share
I think it's caused by azurerm_backup_container_storage_account

### Terraform Configuration Files

```hcl
resource "azurerm_backup_container_storage_account" "protection-container" {
  resource_group_name = azurerm_resource_group.rg.name
  recovery_vault_name = azurerm_recovery_services_vault.vault.name
  storage_account_id  = azurerm_storage_account.sa.id
}

resource "azurerm_backup_protected_file_share" "share1" {
  resource_group_name       = azurerm_resource_group.rg.name
  recovery_vault_name       = azurerm_recovery_services_vault.vault.name
  source_storage_account_id = azurerm_backup_container_storage_account.protection-container.storage_account_id
  source_file_share_name    = azurerm_storage_share.example1.name
  backup_policy_id          = azurerm_backup_policy_file_share.example.id
}

resource "azurerm_backup_protected_file_share" "share2" {
  resource_group_name       = azurerm_resource_group.rg.name
  recovery_vault_name       = azurerm_recovery_services_vault.vault.name
  source_storage_account_id = azurerm_backup_container_storage_account.protection-container.storage_account_id
  source_file_share_name    = azurerm_storage_share.example2.name
  backup_policy_id          = azurerm_backup_policy_file_share.example.id
}
```

### Expected Behaviour

To explain: azurerm_backup_container_storage_account and azurerm_backup_protected_file_share.share1  already exist and azure file share backup is working.
When adding the second "azurerm_backup_protected_file_share" for share 2, the expected behaviour is: 

Enabling azure file share "share2" backup for my share "example2" (using same policy as above)

### Actual Behaviour

Backup is not enabled for share2, azure API answer that the file share is not found and to check if storage account is registered to backup vault (what is already done by "azurerm_backup_container_storage_account" for the first file share) 
It seems that registration is needed for each file share instead of the storage account. If the file share is created after storage account registration to the vault we are unable to enable it for backup.

it seems that every time a file share is added the registration need to be - at least - updated...

```hcl
Error: [ERROR] fileshare 'share2' not found in protectable or protected fileshares, make sure Storage Account "sa" is registered with Recovery Service Vault "vault" (Resource Group "rg")
azurerm_backup_protected_file_share.share2,
on storageaccount_backup.tf line 76, in resource "azurerm_backup_protected_file_share" "share2":
resource "azurerm_backup_protected_file_share" "share2" {
```

### Steps to Reproduce

To reproduce this issue you should: 
1. Create a storage account with a file share, Register it to RSV and enable File share Backup 
2. Run "terraform apply" to create everything
3. Create new file share in the same storage account (already registered to the vault) and enable it for backup
4. Run "terraform apply": you should encounter the error "share not registered to the account"

### Important Factoids
N/A

### References

Other user encounter the same problem but not create issue in the right place or defined it as bug. 
You can find the corresponding issue here:
* https://github.com/hashicorp/terraform/issues/29795
* https://github.com/hashicorp/terraform-provider-azurerm/issues/13841

**Comment by G0PINATH-WW** [comment]
This is fixed in v2.86.0 https://github.com/hashicorp/terraform-provider-azurerm/issues/11184#issuecomment-950677650

**Comment by pier62350** [comment]
Yes it's working fine.
Issue can be closed :)

**Comment by pier62350** [comment]
This functionality has been released in [v2.86.0 of the Terraform Provider](https://github.com/hashicorp/terraform-provider-azurerm/blob/v2.86.0/CHANGELOG.md). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
For further feature requests or bug reports with this functionality, please create a [new GitHub issue](https://github.com/hashicorp/terraform-provider-azurerm/issues/new/choose) following the template. Thank you!
